src/Button/Button.js

```diff
@@ -71,34 +81,104 @@ const StyledButton = createComponent({
+      background: ${outline ? 'transparent' : backgroundColor};
+      border-color: ${outline ? backgroundColor : borderColor || backgroundColor};
+      border-style: solid;
+      border-width: 1px;
       transition: 175ms;
+      white-space: nowrap;
+      outline: none;
+      user-select: none;
 
-      ${loading && loadingCss({ height, fontColor, outline, backgroundColor })};
+      ${icon &&
+        hasText &&
+        css`
+          padding-left: 8px;
+        `}
+
+      ${loading && loadingCss({ height, fontColor, backgroundColor, borderColor, outline })};
 
       &:hover {
-        background: ${outline ? 'transparent' : lighten(0.05, backgroundColor)};
-        border-color: ${lighten(0.05, backgroundColor)};
+        background-color: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${hover &&
+          css`
+            background-color: ${hover.backgroundColor || backgroundColor};
+            border-color: ${hover.borderColor || hover.backgroundColor || backgroundColor};
+            color: ${hover.fontColor || fontColor};
+          `}
       }
 
       &:active {
-        background: ${outline ? 'transparent' : lighten(0.075, backgroundColor)};
-        border-color: ${lighten(0.075, backgroundColor)};
+        background: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${active &&
+          css`
+            background-color: ${active.backgroundColor || backgroundColor};
+            border-color: ${active.borderColor || active.backgroundColor || backgroundColor};
+            color: ${active.fontColor || fontColor};
+          `}
       }
 
+      &[disabled] {
+        pointer-events: none;
+        background: ${backgroundColor};
+        border-color: ${borderColor || backgroundColor};
+        color: ${fontColor};
+
+        ${disabledState &&
+          css`
+            background-color: ${disabledState.backgroundColor || backgroundColor};
+            border-color: ${disabledState.borderColor || disabledState.backgroundColor || backgroundColor};
+            color: ${disabledState.fontColor || fontColor};
+          `}
+      }
+
+      ${text &&
+        css`
+          background-color: transparent !important;
+          border-color: transparent !important;
+          color: ${textColor || backgroundColor} !important;
+        `}
+
       ${space};
     `;
   },
 });
 
-const Button = React.forwardRef((props, ref) => <StyledButton {...props} ref={ref} />);
+const ButtonIcon = styled(Icon)`
+  ${p => css`
+    margin-right: ${p.hasText ? '4px' : 0};
+    margin-top: 2px;
+    font-size: ${p.theme.buttonIconSizes[p.sizing]}px;
+  `}
+`;
+
+const Button = React.forwardRef((props, ref) => {
+  const hasText = !!props.children;
+
+  return (
+    <StyledButton {...props} ref={ref} hasText={hasText}>
+      {props.icon ? (
+        <Flex alignItems="center">
```

Comment:
Is there any harm in including Flex component by default and making just the `<ButtonIcon/>` conditional? Then we could skip the ternary and the `<Flex/>` seems like it could come it handy.